drivers/interrupt_controller/intc_plic.c

```diff
@@ -282,14 +282,19 @@ static int plic_init(const struct device *dev)
 		irq_enable(DT_INST_IRQN(n));                                                       \
 	}
 
+#define PLIC_INTC_INIT_TRIG_0(n) 0
+#define PLIC_INTC_INIT_TRIG_1(n) DT_INST_REG_ADDR_BY_NAME(n, trig)
+#define PLIC_INTC_INIT_TRIG(n) \
+	_CONCAT(PLIC_INTC_INIT_TRIG_, DT_INST_PROP(n, support_edge_interrupt))(n)
+
```

Comment:
This is weird, can you do something like:
```c
CODE_COND_1(DT_INST_REG_HAS_IDX(..., ...), DT_INST_REG_ADDR_BY_NAME(..., ...), (0))
``` 
?

Reply:
These registers offsets are now defined in the driver. Related stuffs in the DT has been removed